src/index.ts

```diff
@@ -105,15 +112,43 @@ class NixInstallerAction {
     this.reinstall = inputs.getBool("reinstall");
     this.startDaemon = inputs.getBool("start-daemon");
     this.trustRunnerUser = inputs.getBool("trust-runner-user");
+    this.runnerOs = process.env["RUNNER_OS"];
   }
 
-  async detectAndForceDockerShim(): Promise<void> {
-    const runnerOs = process.env["RUNNER_OS"];
+  async main(): Promise<void> {
+    await this.detectAndForceDockerShim();
+    await this.install();
+  }
+
+  async post(): Promise<void> {
+    await this.cleanupDockerShim();
+    await this.reportOverall();
+  }
+
+  private get isMacOS(): boolean {
+    return this.runnerOs === "macOS";
+  }
+
+  private get isLinux(): boolean {
+    return this.runnerOs === "Linux";
+  }
+
+  private get isRunningInAct(): boolean {
+    return process.env["ACT"] !== undefined && !(process.env["NOT_ACT"] === "");
+  }
+
+  private get isRunningInNamespaceRunner(): boolean {
+    return (
+      process.env["NSC_VM_ID"] !== undefined &&
+      !(process.env["NOT_NAMESPACE"] === "true")
+    );
+  }
 
+  async detectAndForceDockerShim(): Promise<void> {
     // Detect if we're in a GHA runner which is Linux, doesn't have Systemd, and does have Docker.
     // This is a common case in self-hosted runners, providers like [Namespace](https://namespace.so/),
     // and especially GitHub Enterprise Server.
```

Comment:
nit: I hurt myself in confusion because of this comment: can we move it outside of the `detectAndForceDockerShim` function? Otherwise, it looks like it's talking about the following branch (`!this.isLinux`) which is super confusing (the comment says "which is Linux", but the branch basically says "which is not Linux")

Reply:
Done